tests/core/factory/test_checkpoint_factory.py

```diff
@@ -237,6 +237,56 @@ def _test_checkpoint_factory_delete_success(context):
         context.checkpoints.get(name)
 
 
+@pytest.mark.unit
+def test_checkpoint_factory_get_all_ephemeral():
+    context = get_context(mode="ephemeral")
+    _test_checkpoint_factory_get_all(context)
+
+
+@pytest.mark.filesystem
+def test_checkpoint_factory_get_all_filesystem(empty_data_context):
+    _test_checkpoint_factory_get_all(empty_data_context)
+
+
+@pytest.mark.cloud
+def test_checkpoint_factory_get_all_cloud(empty_cloud_context_fluent):
+    _test_checkpoint_factory_get_all(empty_cloud_context_fluent)
```

Comment:
`non-blocking`
You can apply marks dynamically which would allow you to have 1 parametrized test.
https://stackoverflow.com/a/73556599/6304433

Reply:
Oh good idea. Thanks!